genai/example_test.go

```diff
@@ -258,6 +258,25 @@ func ExampleClient_ListModels() {
 	}
 }
 
+func ExampleFunctionSchema() {
+	// Define a function to use as a tool.
+	weatherToday := func(city string) string {
+		return "comfortable, if you have the right clothing"
+	}
+	// You can construct the Schema for this function by hand, or
+	// let Go reflection figure it out.
+	// Reflection can't see parameter names, so provide those too.
+	schema, err := genai.FunctionSchema(weatherToday, "city")
+	if err != nil {
+		// Not every type can be used in a tool function.
+		panic(err)
+	}
+
+	// Use the schema to create a FunctionDeclaration.
+	// Use that to populate Model.Tools.
```

Comment:
Why not continue to show how to do all of this? 
We don't have other samples for tools, AFAICS